### PR TITLE
feat(rest) : Rest Api for ECC page

### PR DESCRIPTION
**Pull request by keerthi-bl**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Issue: 
#2036 

http://localhost:8080/resource/api/ecc?page=0&page_entries=3&sort=name%2Cdesc

Sample response : 

{
    "_embedded": {
        "sw360:jSONObjects": [
            {
                "assessor_contact_person": "",
                "name": "compfeb2 (50)",
                "DT_RowId": "d29b5995f30146e5ad30b84a35f42112",
                "id": "d29b5995f30146e5ad30b84a35f42112",
                "version": "50",
                "assessor_dept": "",
                "status": "OPEN",
                "group": "",
                "assessment_date": ""
            },
            {
                "assessor_contact_person": "",
                "name": "Test Compo 2023 1 (22.23)",
                "DT_RowId": "e06f3a093ca1453085fd27c0017e6e2c",
                "id": "e06f3a093ca1453085fd27c0017e6e2c",
                "version": "22.23",
                "assessor_dept": "",
                "status": "OPEN",
                "group": "",
                "assessment_date": ""
            },
            {
                "assessor_contact_person": "eccadmin@sw360.org",
                "name": "Component5.19 (1.1)",
                "DT_RowId": "2e7f86c8236e4dd299328cbe3f7f7ae0",
                "id": "2e7f86c8236e4dd299328cbe3f7f7ae0",
                "version": "1.1",
                "assessor_dept": "DEPARTMENT",
                "status": "REJECTED",
                "group": "",
                "assessment_date": "2023-02-03"
            }
        ]
    },
    "_links": {
        "first": {
            "href": "http://localhost:8080/resource/api/ecc?page=0&page_entries=3"
        },
        "previous": {
            "href": "http://localhost:8080/resource/api/ecc?page=0&page_entries=3"
        },
        "last": {
            "href": "http://localhost:8080/resource/api/ecc?page=0&page_entries=3"
        },
        "curies": [
            {
                "href": "http://localhost:8080/resource/docs/{rel}.html",
                "name": "sw360",
                "templated": true
            }
        ]
    },
    "page": {
        "size": 3,
        "totalElements": 3,
        "totalPages": 1,
        "number": 1
    }
}

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by keerthi-bl** [comment]
Merge conflicts resolved

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/ecc/SW360EccService.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    private final RestControllerHelper restControllerHelper;
+    
+    // ECC release view datatables, index of columns
+    private static final String RELEASE_DT_ROW_ECC_STATUS = "status";
+    private static final String RELEASE_DT_ROW_NAME = "name";
+    private static final String RELEASE_DT_ROW_VERSION = "version";
+    private static final String RELEASE_DT_ROW_GROUP = "group";
+    private static final String RELEASE_DT_ROW_ASSESSOR_CONTACT_PERSON = "assessor_contact_person";
+    private static final String RELEASE_DT_ROW_ASSESSOR_DEPARTMENT = "assessor_dept";
+    private static final String RELEASE_DT_ROW_ASSESSMENT_DATE = "assessment_date";
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/ecc/SW360EccService.java:69]: No need to create the above variables, instead use struct EccInformation{} of components.thrift

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/ecc/SW360EccService.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+    public JSONArray getECCPageData(User user,Pageable pageable) throws TException {
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/ecc/SW360EccService.java:71]: Use already created Pagination methods present in restControllerhelper

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/ecc/SW360EccService.java [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        return releaseSummary;
+    }
+
+    private JSONArray getReleaseData(List<Release> releaseList, Pageable pageable) {
+        List<Release> sortedReleases = sortReleaseList(releaseList, pageable);
+        long count = getReleaseDataCount(pageable, releaseList.size());
+        final int start = 0;
+        JSONArray releaseData = new JSONArray();
+        for (int i = start; i < count; i++) {
+            JSONObject jsonObject = new JSONObject();
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/ecc/SW360EccService.java:100]: No need to use jsonObject, instead use eccInformation field of releases after iterating the release list

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/ecc/SW360EccService.java [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        case RELEASE_DT_ROW_ASSESSMENT_DATE:
+            Collections.sort(releaseList, compareByAssessmentDate(isAsc));
+            break;
+        default:
+            break;
+        }
+        return releaseList;
+    }
+
+    private static Comparator<Release> compareByECCStatus(boolean isAscending) {
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/ecc/SW360EccService.java:156]: Create comparator method in ResourceComparatorGenerator.java

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/ecc/SW360EccController.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
+/*
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/ecc/SW360EccController.java:1]: Change filename to EccController.java and update the licenseHeader year

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/ecc/SW360EccService.java [hunk 6]
```diff
@@ -0,0 +1,212 @@
+/*
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/ecc/SW360EccService.java:1]: Change filename to EccService.java and update licenseHeader year in newly created files.

**Comment by rudra-superrr** [comment]
Testing was successful.
image

**Comment by rudra-superrr** [comment]
One thing, request structure and response structure fields are not shown in api docs.

**Comment by keerthi-bl** [comment]
Review comment addressed.

**Comment by rudra-superrr** [comment]
Code looks good.

**Pull request merged**